File: Problems 0000-0099/57. Insert Interval/solution.cpp
class Solution {
    public:
        vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
            int n = intervals.size();
            int i = 0;
            vector<vector<int>> res;
    
            while (i < n && intervals[i][1] < newInterval[0]) {
                res.push_back(intervals[i++]);
            }
    
            int start = newInterval[0];
            int end = newInterval[1];
            while (i < n && intervals[i][0] <= newInterval[1]) {
                start = min(intervals[i][0], start);
                end = max(intervals[i][1], end);
                i++;
            }
            res.push_back({start, end});
    
            while (i < n) {
                res.push_back(intervals[i++]);
            }
            return res;
        }
    };

File: Problems 0000-0099/57. Insert Interval/solution.java
class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        int n = intervals.length;
        int i = 0;
        int index = 0;
        int[][] res = new int[n + 1][2];
        while (i < n && intervals[i][1] < newInterval[0]) {
            res[index++] = intervals[i++];
        }

        res[index++] = newInterval;

        while (i < n && intervals[i][0] <= newInterval[1]) {
            if (intervals[i][0] < newInterval[0]) {
                newInterval[0] = intervals[i][0];
            }
            if (intervals[i][1] > newInterval[1]) {
                newInterval[1] = intervals[i][1];
            }
            i++;
        }

        while (i < n) {
            res[index++] = intervals[i++];
        }

        return Arrays.copyOf(res, index);
    }
}